dodiscover/metrics.py

```diff
@@ -80,3 +80,69 @@ def confusion_matrix_networks(
     # compute the confusion matrix
     conf_mat = confusion_matrix(y_true, y_pred, labels=labels, normalize=normalize)
     return conf_mat
+
+
+def structure_hamming_dist(
+    true_graph: Union[nx.Graph, nx.DiGraph],
```

Comment:
We define this union in a bunch of places - it may be worth explicitly defining a type that captures it in a common `typings.py` module: e.g. `GraphInput = Union[nx.Graph, nx.DiGraph]`

Reply:
Done, but instead called it `NetworkxGraph` for now to differentiate what is in Networkx vs not.